### PR TITLE
Homework3: Katsiaryna Zabelava

### PR DESCRIPTION
**Pull request by Tobichimaru**

### CODE REVIEW

File: timer_thread.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+timer = threading.Timer(0.1, stub_func)
+
+
+def odd():
+    global curr, end
+    while True:
+        if curr >= end:
+            break
+        if curr % 2 != 0:
```
Comment (alenaPy) [inline timer_thread.py]: `if curr % 2:`

File: semaphore_thread.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+semaphore = threading.Semaphore()
+start = 1
+end = 100
+curr = start
+
+
+def odd():
+    global curr, semaphore, end
+    while True:
+        if curr % 2 == 1:
```
Comment (alenaPy) [inline semaphore_thread.py]: `if curr % 2:`

File: lock_thread.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+def even():
+    global curr, lock, end
+    while True:
+        if curr % 2 == 0:
+            lock.acquire()
+            if curr > end:
+                break
+            print(curr)
+            curr = curr + 1
```
Comment (alenaPy) [inline lock_thread.py]: `curr += 1' pythonic way

**Comment by Tobichimaru** [comment]
The PR updated according to the comments, thank you for the review

**Pull request closed without merging**